### PR TITLE
Makes it so assistants can assign themselves to departments.

### PR DESCRIPTION
**Pull request by praisenarsie**
also makes me gm

### CODE REVIEW

File: code/modules/jobs/job_types/assistant.dm [hunk 1]
```diff
@@ -22,6 +22,53 @@ Assistant
 	else
 		return ..()
 
+GLOBAL_LIST_INIT(available_depts, list(ASS_DEPT_ENGINEERING, ASS_DEPT_MEDICAL, ASS_DEPT_SCIENCE, ASS_DEPT_SUPPLY))
+
+/datum/job/assistant/after_spawn(mob/living/carbon/human/H, mob/M) // this is pretty much copied from security officer
+	// Assign department
+	var/department
+	if(M && M.client && M.client.prefs)
+		department = M.client.prefs.prefered_security_department
```
Comment (cyclowns) [inline code/modules/jobs/job_types/assistant.dm]: change to M.client.prefs.preferred_assistant_department

File: code/modules/client/preferences.dm [hunk 2]
```diff
@@ -71,6 +71,7 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 
 	var/list/custom_names = list("human", "clown", "mime", "ai", "cyborg", "religion", "deity")
 	var/prefered_security_department = SEC_DEPT_RANDOM
```
Comment (cyclowns) [inline code/modules/client/preferences.dm]: change prefered to preferred

**Comment by praisenarsie** [comment]
travis is gonna fail but i'll fix my shit tomorrow

File: code/modules/jobs/job_types/assistant.dm [hunk 3]
```diff
@@ -22,6 +22,53 @@ Assistant
...
+		if(!LAZYLEN(GLOB.available_depts) || department == "None")
+			return
+		else if(department in GLOB.available_depts)
+			LAZYREMOVE(GLOB.available_depts, department)
+		else
+			department = pick_n_take(GLOB.available_depts)
+	var/ears = null
+	var/accessory = null
+	var/list/dep_access = null
```
Comment (cyclowns) [inline code/modules/jobs/job_types/assistant.dm]: department access is never added to their ID?

**Comment by praisenarsie** [comment]
features later

**Pull request closed without merging**